index.html

```diff
@@ -543,6 +543,111 @@ <h6>JWT Decoding</h6>
       </section>
     </section>
 
+    <section id="production">
+      <h2>Production</h2>
+
+      <p>
+        This section describes how to produce a VC-JWT encoded
+        <code>VerifiableCredential</code> from a <code>Credential</code>.
+      </p>
+
+      <p class="advisment">
+        There are currently 2 competing solutions to this problem described below. It is a goal
+        of the v2 work to resolve them and
+        reduce production rules to a single, simple, set of instructions that
+        any implementer can easily meet if they possess a software library
+        supporting [[RFC7515]] or [[RFC7519]].
+      </p>
+
+      <section>
+        <h3>In addition to...</h3>
+
+        <p>
+          There are several members (claims) of the
+          <code>Credential</code> which will need to be translated to their JOSE
+          form, and included next to the <code>vc</code> or
+          <code>vp</code> member in the <code>payload</code>.
+        </p>
+
+        <p>
+          If a member is not present in the <code>Credential</code> it MUST NOT
+          be present in the <code>VerifiableCredential</code> as either a claim
+          in the payload or a claim in the <code>vc</code> attribute of the
+          payload.
+        </p>
+
+        <p>
+          We start with an empty header, and payload objects, and we add members
```

Comment:
Shouldn't the header always include at least the "alg" value used to sign with and probably the "kid"?  We're including those, not because of the contents of the credential, but as part of securing the JWT.

Reply:
Good question. At this point, its not known what the header will be!

We add the `alg` based on the key being used to protect the header and payload.

In most implementations I have seen, this happens either automatically at the time of signing, or just before it, based on the private key or signer passed to the library.... Because `alg` is optional in the JWK spec, sometimes it needs to be passed as an additional argument, next to the private key... for example:

```js
const jwt = await new jose.SignJWT({ 'urn:example:claim': true })
  .setProtectedHeader({ alg: 'ES256' }) // required here, since we don't know what type of key will be used.
  .setIssuedAt()
  .setIssuer('urn:example:issuer')
  .setAudience('urn:example:audience')
  .setExpirationTime('2h')
  .sign(privateKey) // alg might not be known here, this is just a generic private key... not a JWK.

console.log(jwt)
```

https://github.com/panva/jose/blob/HEAD/docs/classes/jwt_sign.SignJWT.md#readme